### PR TITLE
Change clojure.algo.generic/fmap to cats.core/fmap

### PR DESCRIPTION
**Pull request by szdavid92**

### CODE REVIEW

File: sre/src/sre/plan/op.clj [hunk 1]
```diff
@@ -18,7 +18,7 @@
 (extend-type Op
   IBind
   (bind-map [this var-map]
-    (let [replace (partial fmap (fn [v] (fmap (fn [p] (fmap #(var-map %1) p)) v)))]
+    (let [replace (partial fmap (fn [v] (println v)(fmap (fn [p] (println "p" p)(fmap #(do (println %) (var-map %)) p)) v)))]
```
Comment (szarnyasg) [inline sre/src/sre/plan/op.clj]: why println?

Comment (szdavid92) [reply sre/src/sre/plan/op.clj]: sheet, i left it while debugging. nice catch!

**Comment by szdavid92** [comment]
Cannot merge, there is a bug with cats, which seems to bi fixed in an unrelease 2.2.1. See https://github.com/funcool/cats/issues/209

**Comment by szdavid92** [comment]
blocked by funcool/cats#209

**Comment by szarnyasg** [comment]
> which seems to be fixed in an unreleased 2.2.1.
That's actually my conjecture... https://github.com/FTSRG/cheat-sheets/wiki/Engineering-laws

> (Gabor Szarnyas) My conjecture: "When you start using an open-source software project, you soon encounter a problem that is very difficult to tackle with the current stable version, but will have an elegant, 'out-of-the-box' solution in one of the project's future releases. This is usually promised in the roadmap of the projects or just mentioned casually on the issue ticket that you found."

**Comment by szdavid92** [comment]
nah,, its a bug and actually fixed, just not yet released :D

File: sre/src/sre/plan/pattern.clj [hunk 2]
```diff
@@ -101,7 +100,7 @@
     (try
       (let [[inner-constr inner-op inner-weight inner-task] (compile pattern config compiler-opts)
             ;; TODO find good weight
-            weight 0.5
+            weight 1
```
Comment (szdavid92) [inline sre/src/sre/plan/pattern.clj]: unintended change

**Comment by szarnyasg** [comment]
:relieved:

**Comment by szdavid92** [comment]
fixes #268

**Pull request closed without merging**